vcu/Src/canReceive.c

```diff
@@ -170,9 +174,24 @@ void CAN_Msg_MC_Read_Write_Param_Response_Callback() // parameter message
 
 void CAN_Msg_MC_Fault_Codes_Callback() // 100 hz, broadcast message
 {
+    static uint8_t numCriticalFaults = 0;
     // Each bit represents a fault
     // Combine them to be sent over DTCs
     inverterFaultCode = INV_Post_Fault_Hi | INV_Post_Fault_Lo | INV_Run_Fault_Hi | INV_Run_Fault_Lo;
+
+    if(inverterFaultCode & INVERTER_FATAL_FAULT_MASK) {
+        fsmSendEventUrgentISR(&fsmHandle, EV_Inverter_Fault);
+    }
+    else if(inverterFaultCode & INVERTER_CRITICAL_FAULT_MASK) {
+        numCriticalFaults++;
+        if(numCriticalFaults == 3) {
+            fsmSendEventUrgentISR(&fsmHandle, EV_Inverter_Fault);
+        }
+        else {
+            mcClearFaults();
```

Comment:
if it clears the fault successfully, don't we want to decrement _numCriticalFaults_?